### ISSUE TITLE
Flutter project unable to run on iOS simulator but able to run on physical phone.

### ISSUE THREAD

**Issue by brycegoh** [flutter/flutter#89938]
## Steps to Reproduce

Update 13/09/2021:
Downgrading flutter to v2.2.3 has solved my issues. Able to run on simulator with no errors.

Currently on m1 MacBook Air
Flutter on latest stable channel which is version 2.5.0

1. Run `flutter run --flavor dev`.

**Expected results:** 
Flutter runs app on IOS simulator on dev mode.

**Actual results:** 

`flutter run --flavor dev -v`
```
output:
                            libopencore-amrnb -framework libsndfile -framework libswresample -framework libswscale
-framework libvorbis
                            -framework libvorbisenc -framework libvorbisfile -framework libwebp -framework mobileffmpeg
-framework nanopb
                            -framework opus -framework package_info_plus -framework path_provider -framework
permission_handler -framework
                            sentry_flutter -framework share -framework shine -framework soxr -framework speex -framework
sqflite -framework
                            store_redirect -framework stripe_ios -framework tau_core -framework twolame -framework
url_launcher -framework
                            vibration -framework vo-amrwbenc -framework wavpack -framework webview_flutter -weak_framework
UserNotifications
                            -Xlinker -sectcreate -Xlinker __TEXT -Xlinker __entitlements -Xlinker

/Users/brycegoh/Library/Developer/Xcode/DerivedData/Runner-ananiohtlsfwecgddlajixutvbpg/Build/Intermediates.noindex
                            /Runner.build/Debug-dev-iphonesimulator/Runner.build/Runner.app-Simulated.xcent -framework
Pods_Runner -Xlinker
                            -no_adhoc_codesign -Xlinker -dependency_info -Xlinker

/Users/brycegoh/Library/Developer/Xcode/DerivedData/Runner-ananiohtlsfwecgddlajixutvbpg/Build/Intermediates.noindex

/Runner.build/Debug-dev-iphonesimulator/Runner.build/Objects-normal/x86_64/Runner_dependency_info.dat -o

/Users/brycegoh/Library/Developer/Xcode/DerivedData/Runner-ananiohtlsfwecgddlajixutvbpg/Build/Intermediates.noindex
                            /Runner.build/Debug-dev-iphonesimulator/Runner.build/Objects-normal/x86_64/Binary/Runner
                    
                    
[ +779 ms] Could not build the application for the simulator.
[   +6 ms] Error launching application on iPhone 12.
[  +15 ms] "flutter run" took 33,546ms.
[  +12 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:669:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1125:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command_runner.dart:288:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:236:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:92:3)
           <asynchronous suspension>
           
           
[ +216 ms] ensureAnalyticsSent: 211ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
```

`flutter doctor -v`
output:
```
[✓] Flutter (Channel stable, 2.5.0, on macOS 11.5.2 20G95 darwin-arm, locale en-SG)
    • Flutter version 2.5.0 at /Users/brycegoh/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cc385b4b8 (4 days ago), 2021-09-07 23:01:49 -0700
    • Engine revision f0826da7ef
    • Dart version 2.14.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/brycegoh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.0

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google
    Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (1 available)
    • iPhone 12 (mobile) • FD840261-179D-4455-B369-AABBC8B653FC • ios • com.apple.CoreSimulator.SimRuntime.iOS-14-5
      (simulator)

! Doctor found issues in 1 category.
```

**Comment by Vidtion** [comment]
same issue on intel MacBook pro with flutter stable 2.5.0

**Comment by darshankawar** [comment]
> 1. flutter run --flavor dev

@brycegoh 
I see that you are trying to run a flavor app. I tried a sample app using [this](https://medium.com/@animeshjain/build-flavors-in-flutter-android-and-ios-with-different-firebase-projects-per-flavor-27c5c5dac10b) article and ran it on stable 2.2.3 and on stable 2.5.0, and on both, worked properly without any issues.

I ran the command: `flutter run --flavor dev -t lib/main_dev.dart` and ran it on iOS simulator:


<img width="371" alt="Screenshot 2021-09-13 at 4 38 12 PM" src="https://user-images.githubusercontent.com/67046386/133074083-d09a62c2-8bed-43e4-85f0-1693a93aba40.png">




```
Launching lib/main_dev.dart on iPhone 12 Pro Max in debug mode...
Running Xcode build...                                                  
 └─Compiling, linking and signing...                         8.7s
Xcode build done.                                           29.6s
Syncing files to device iPhone 12 Pro Max...                        77ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
```

You may check that article and see if you are missing anything at your end.
If issue still persists, provide a complete `verbose` log of the error.

**Comment by Vidtion** [comment]
> > 1. flutter run --flavor dev
> 
> @brycegoh
> I see that you are trying to run a flavor app. I tried a sample app using [this](https://medium.com/@animeshjain/build-flavors-in-flutter-android-and-ios-with-different-firebase-projects-per-flavor-27c5c5dac10b) article and ran it on stable 2.2.3 and on stable 2.5.0, and on both, worked properly without any issues.
> 
> I ran the command: `flutter run --flavor dev -t lib/main_dev.dart` and ran it on iOS simulator:
> 
> <img alt="Screenshot 2021-09-13 at 4 38 12 PM" width="371" src="https://user-images.githubusercontent.com/67046386/133074083-d09a62c2-8bed-43e4-85f0-1693a93aba40.png">
> 
> ```
> Launching lib/main_dev.dart on iPhone 12 Pro Max in debug mode...
> Running Xcode build...                                                  
>  └─Compiling, linking and signing...                         8.7s
> Xcode build done.                                           29.6s
> Syncing files to device iPhone 12 Pro Max...                        77ms
> 
> Flutter run key commands.
> r Hot reload. 🔥🔥🔥
> ```
> 
> You may check that article and see if you are missing anything at your end.
> If issue still persists, provide a complete `verbose` log of the error.

I got `BUILD FAILED` on flutter 2.5.0, and `BUILD SUCCEEDED` on flutter 2.2.3

```
            ** BUILD FAILED **
            The following build commands failed:
            	CompileSwift normal x86_64
            	CompileSwift normal i386
            	CompileSwift normal arm64
            	Ld /Users/coin/Library/Developer/Xcode/DerivedData/Runner-bpolijzkaufkhoeymhbdjebbmltz/Build/Intermediates.noindex/Pods.build/Debug-pgy-iphonesimulator/flutter_nim.build/Objects-normal/arm64/Binary/flutter_nim normal arm64
            (4 failures)
[ +826 ms] Xcode build done.                                           75.6s
[   +1 ms] Failed to build iOS app
[        ] Error output from Xcode build:
           ↳
[        ]     ** BUILD FAILED **
               The following build commands failed:
               	CompileSwift normal x86_64
               	CompileSwift normal i386
               	CompileSwift normal arm64
               	Ld /Users/coin/Library/Developer/Xcode/DerivedData/Runner-bpolijzkaufkhoeymhbdjebbmltz/Build/Intermediates.noindex/Pods.build/Debug-pgy-iphonesimulator/flutter_nim.build/Objects-normal/arm64/Binary/flutter_nim normal arm64
               (4 failures)
[        ] Xcode's output:
           ↳
[  +84 ms]     Command line invocation:
                   /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -configuration Debug-pgy VERBOSE_SCRIPT_LOGGING=YES -workspace Runner.xcworkspace -scheme pgy BUILD_DIR=/Users/coin/Documents/gits/workbench/build/ios -sdk iphonesimulator FLUTTER_SUPPRESS_ANALYTICS=true COMPILER_INDEX_STORE_ENABLE=NO
               User defaults from command line:
                   IDEPackageSupportUseBuiltinSCM = YES
               Build settings from command line:
                   BUILD_DIR = /Users/coin/Documents/gits/workbench/build/ios
                   COMPILER_INDEX_STORE_ENABLE = NO
                   FLUTTER_SUPPRESS_ANALYTICS = true
                   SDKROOT = iphonesimulator14.5
                   VERBOSE_SCRIPT_LOGGING = YES
               note: Using new build system
               note: Building targets in parallel
               note: Planning build
               note: Analyzing workspace
               note: Constructing build description
               note: Build preparation complete
```
```
[✓] Flutter (Channel stable, 2.5.0, on macOS 11.4 20F71 darwin-x64, locale zh-Hans-CN)
    • Flutter version 2.5.0 at /Users/coin/Documents/flutter-sdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cc385b4b8 (6 days ago), 2021-09-07 23:01:49 -0700
    • Engine revision f0826da7ef
    • Dart version 2.14.0
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/coin/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/coin/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7665

[✓] IntelliJ IDEA Community Edition (version 2021.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7665

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (2 available)
    • iPhone 11 Pro Max (mobile) • 7FEA026E-226C-4500-A4CC-00E8FA0F5E6A • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 93.0.4577.63

• No issues found!
```


```
** BUILD SUCCEEDED **
[+1181 ms] Xcode build done.                                           179.5s
[   +2 ms] executing: [/Users/coin/Documents/gits/workbench/ios/] xcrun xcodebuild -configuration Debug-pgy VERBOSE_SCRIPT_LOGGING=YES -workspace Runner.xcworkspace -scheme pgy BUILD_DIR=/Users/coin/Documents/gits/workbench/build/ios -sdk iphonesimulator FLUTTER_SUPPRESS_ANALYTICS=true COMPILER_INDEX_STORE_ENABLE=NO -showBuildSettings
[        ] executing: [/Users/coin/Documents/gits/workbench/ios/] xcrun xcodebuild -configuration Debug-pgy VERBOSE_SCRIPT_LOGGING=YES -workspace Runner.xcworkspace -scheme pgy BUILD_DIR=/Users/coin/Documents/gits/workbench/build/ios -sdk iphonesimulator FLUTTER_SUPPRESS_ANALYTICS=true COMPILER_INDEX_STORE_ENABLE=NO -showBuildSettings
[+3461 ms] Command line invocation:
               /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -configuration Debug-pgy VERBOSE_SCRIPT_LOGGING=YES -workspace Runner.xcworkspace -scheme pgy BUILD_DIR=/Users/coin/Documents/gits/workbench/build/ios -sdk iphonesimulator FLUTTER_SUPPRESS_ANALYTICS=true COMPILER_INDEX_STORE_ENABLE=NO -showBuildSettings
           User defaults from command line:
               IDEPackageSupportUseBuiltinSCM = YES
           Build settings from command line:
               BUILD_DIR = /Users/coin/Documents/gits/workbench/build/ios
               COMPILER_INDEX_STORE_ENABLE = NO
               FLUTTER_SUPPRESS_ANALYTICS = true
               SDKROOT = iphonesimulator14.5
               VERBOSE_SCRIPT_LOGGING = YES
           Build settings for action build and target Runner:
               ACTION = build
               AD_HOC_CODE_SIGNING_ALLOWED = YES
               ALTERNATE_GROUP = staff
               ALTERNATE_MODE = u+w,go-w,a+rX
               ALTERNATE_OWNER = coin
               ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES
               ALWAYS_SEARCH_USER_PATHS = NO
               ALWAYS_USE_SEPARATE_HEADERMAPS = NO
               APPLE_INTERNAL_DEVELOPER_DIR = /AppleInternal/Developer
               APPLE_INTERNAL_DIR = /AppleInternal
               APPLE_INTERNAL_DOCUMENTATION_DIR = /AppleInternal/Documentation
               APPLE_INTERNAL_LIBRARY_DIR = /AppleInternal/Library
               APPLE_INTERNAL_TOOLS = /AppleInternal/Developer/Tools
               APPLICATION_EXTENSION_API_ONLY = NO
               APPLY_RULES_IN_COPY_FILES = NO
               APPLY_RULES_IN_COPY_HEADERS = NO
               ARCHS = x86_64
               ARCHS_STANDARD = arm64 x86_64 i386
               ARCHS_STANDARD_32_64_BIT = arm64 i386 x86_64
               ARCHS_STANDARD_32_BIT = i386
               ARCHS_STANDARD_64_BIT = arm64 x86_64
               ARCHS_STANDARD_INCLUDING_64_BIT = arm64 x86_64 i386
               ARCHS_UNIVERSAL_IPHONE_OS = arm64 i386 x86_64
               ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon
               AVAILABLE_PLATFORMS = appletvos appletvsimulator iphoneos iphonesimulator macosx watchos watchsimulator
```

```

[✓] Flutter (Channel stable, 2.2.3, on macOS 11.4 20F71 darwin-x64, locale zh-Hans-CN)
    • Flutter version 2.2.3 at /Users/coin/Documents/flutter-sdk/flutter
    • Framework revision f4abaa0735 (3 months ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/coin/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/coin/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7665

[✓] IntelliJ IDEA Community Edition (version 2021.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7665

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (2 available)
    • iPhone 11 Pro Max (mobile) • 7FEA026E-226C-4500-A4CC-00E8FA0F5E6A • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 93.0.4577.63

• No issues found!
```

**Comment by TahaTesser** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

@Vidtion
Please file a new issue with all the details, including code sample
